File: fix_master_room.py
import re

with open('src/pages/MasterRoom.tsx', 'r') as f:
    code = f.read()

domain_start = code.find('function DomainSkinControl')
if domain_start != -1:
    first_part = code[:domain_start]
    rest = code[domain_start:]
    
    # find the end of DomainSkinControl
    end_match = re.search(r'  \);\n\}\n', rest)
    if end_match:
        end_idx = end_match.end()
        domain_code = rest[:end_idx]
        remaining = rest[end_idx:]
        
        # we need to put domain_code BEFORE the "function WatermarkEditor" in first_part
        # but wait, first_part ends with "function WatermarkEditor({ clients, setClients }: { clients: Client[], setClients: any }) {\n"
        # Let's fix this safely:
        watermark_start = first_part.rfind('function WatermarkEditor')
        if watermark_start != -1:
            pre_watermark = first_part[:watermark_start]
            watermark_decl = first_part[watermark_start:]
            new_code = pre_watermark + domain_code + '\n' + watermark_decl + remaining
            with open('src/pages/MasterRoom.tsx', 'w') as f:
                f.write(new_code)

File: fix_master_room.js
const fs = require('fs');
let code = fs.readFileSync('src/pages/MasterRoom.tsx', 'utf8');

// Find the nested DomainSkinControl and extract it
const domainStart = code.indexOf('function DomainSkinControl');
const firstPart = code.substring(0, domainStart);
let rest = code.substring(domainStart);

// We know DomainSkinControl ends at `  );\n}\n`
const domainEndMatch = rest.match(/  \);\n\}\n/);
const domainEndIndex = domainEndMatch.index + domainEndMatch[0].length;

const domainSkinControlCode = rest.substring(0, domainEndIndex);
const remainingCode = rest.substring(domainEndIndex);

// Reassemble
const newCode = firstPart.replace('function WatermarkEditor', domainSkinControlCode + '\n\nfunction WatermarkEditor') + remainingCode;

fs.writeFileSync('src/pages/MasterRoom.tsx', newCode);
